lib/bundler/resolver.rb

```diff
@@ -250,28 +250,34 @@ def search_for(dependency)
         if vertex = @base_dg.vertex_named(dependency.name)
           locked_requirement = vertex.payload.requirement
         end
-        if results.any?
-          nested = []
-          results.each do |spec|
-            version, specs = nested.last
-            if version == spec.version
-              specs << spec
-            else
-              nested << [spec.version, [spec]]
+        spec_groups = begin
+          if results.any?
+            nested = []
+            results.each do |spec|
+              version, specs = nested.last
+              if version == spec.version
+                specs << spec
+              else
+                nested << [spec.version, [spec]]
+              end
             end
+            nested.reduce([]) do |groups, (version, specs)|
+              next groups if locked_requirement && !locked_requirement.satisfied_by?(version)
+              groups << SpecGroup.new(specs)
+            end
+          else
+            []
           end
-          nested.reduce([]) do |groups, (version, specs)|
-            next groups if locked_requirement && !locked_requirement.satisfied_by?(version)
-            groups << SpecGroup.new(specs)
-          end
-        else
-          []
         end
+        # MODO: ideally trying to get GVP to sort by major. but some test cases aren't showing consistent order
+        # by default. (so we don't have to have this conditional here)
+        if @gem_version_promoter.level != :major
+          @gem_version_promoter.sort_versions(dependency, spec_groups)
+        else
+          spec_groups
+        end # .tap {|sg| STDERR.puts "sg>> #{sg.inspect}" }
```

Comment:
yeah, sorry - i'm playing with those lines right now - i knew i was coming back to that section next, didn't think about it before pushing that commit.